ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,64 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void addSecretToJfrog(IPropertyTree &item)
+{
+    StringBuffer password;
+    const char *user = item.queryProp("@jfrogUser");
+    getSecretValue(password, "jfrog", user, "password", true);
+    password.stripChar('\n').stripChar(' ').str(); // Strip newlines/whitespaces in case the secret has been entered by hand.
+
+    StringBuffer fileContents;
+    const char *server = item.queryProp("@jfrogServer");
+    fileContents.setf("{\n\
+    \"servers\": [\n\
+        {\n\
+        \"url\": \"%s/\",\n\
+        \"artifactoryUrl\": \"%s/artifactory/\",\n\
+        \"distributionUrl\": \"%s/distribution/\",\n\
+        \"xrayUrl\": \"%s/xray/\",\n\
+        \"missionControlUrl\": \"%s/mc/\",\n\
+        \"pipelinesUrl\": \"%s/pipelines/\",\n\
+        \"user\": \"%s\",\n\
+        \"password\": \"%s\",\n\
+        \"serverId\": \"HPCC\",\n\
+        \"isDefault\": true\n\
+        }\n\
+    ],\n\
+    \"version\": \"6\"\n}\n", server, server, server, server, server, server, user, password.str());
+
+    StringBuffer configPath(getenv("HOME"));
+    atomicWriteFile(configPath.append(PATHSEPCHAR).append(".jfrog").append(PATHSEPCHAR).append("jfrog-cli.conf.v6"), fileContents);
+}
+
+void getResourceFromJfrog(StringBuffer &localPath, IPropertyTree &item)
+{
+    addSecretToJfrog(item);
```

Comment:
security: Ensure this file is deleted when the command has executed (for success or failure)

Reply:
Fixed.